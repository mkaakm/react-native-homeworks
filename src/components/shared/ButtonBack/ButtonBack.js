import React from "react";
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';

import {COLORS} from "../../../color";

export default function Button({text, style, onPress}) {

    return (
        <View style={[styles.backButton, style]}>
            <TouchableOpacity onPress={onPress}>
                <Text style={styles.buttonText}>{text}</Text>
            </TouchableOpacity>
        </View>
    );
}

const styles = StyleSheet.create({
    backButton: {
        color: COLORS.secondary,
        borderColor: COLORS.secondary,
        borderRadius: 15,
        backgroundColor: COLORS.secondary,
        textAlign:'center',
        borderStyle:'solid',
        borderWidth: 1,
        display: 'flex',
        width: 50,
        paddingTop: 5,
        paddingBottom: 5,
        position:'absolute',
        top: 20,
        left: 10
    },
    buttonText:{
        textAlign:'center'
    }
});
