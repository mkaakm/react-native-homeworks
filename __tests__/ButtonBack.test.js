import React from 'react';
import renderer from 'react-test-renderer';
import ButtonBack from '../src/shared/ButtonBack'
import {StyleSheet} from 'react-native';


test('renders correctly', () => {
    const tree = renderer.create(<ButtonBack style={styles.buttonBack}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

const styles = StyleSheet.create({
    buttonBack:{
        textAlign:"center"
    }
})
