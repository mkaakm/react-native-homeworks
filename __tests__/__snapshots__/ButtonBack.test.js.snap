// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
        "borderColor": "red",
        "borderRadius": 15,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "red",
        "display": "flex",
        "left": 10,
        "paddingBottom": 5,
        "paddingTop": 5,
        "position": "absolute",
        "textAlign": "center",
        "top": 20,
        "width": 50,
      },
      Object {
        "textAlign": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  />
</View>
`;
